### PR TITLE
add array bracket notation support

### PR DESCRIPTION
**Pull request by mreinstein**
this works now:
```javascript
const obj = { a: [ { b: 'cool' } ]};
const val = dotProp.get(obj, 'a[0].b'); // val === 'cool'
```

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ test('get', t => {
 	F4Class.prototype.foo = 1;
 	const f4 = new F4Class();
 	t.is(dotProp.get(f4, 'foo'), 1); // #46
+
+	const fixture3 = {a: [{b: 'cool'}]};
+	t.is(dotProp.get(fixture3, 'a[0].b'), 'cool');
+	t.is(dotProp.get(fixture3, 'a[1].b', 'defaultVal'), 'defaultVal');
```
Comment (sindresorhus) [inline test.js:61]: Try to come up with some more assertions. For example, `a[0].nonexistant` with and without default value.

**Comment by sindresorhus** [comment]
Also see https://github.com/sindresorhus/dot-prop/issues/71

**Comment by sindresorhus** [comment]
This will need docs and more tests.

**Comment by mreinstein** [comment]
> Also see #71

Yeah, I saw that when looking through back issues to see if this was brought up anywhere. The title of 71 matches what this PR does, but the content of the issue seems somewhat unrelated (claims to backwards incompatibility breaks and other stuff)

> This will need docs and more tests (for each method).

I didn't do that in the initial PR because all of the public methods (`get`, `set`, `has`) use the same underlying ` getPathSegments` to resolve the path, so they seemed redundant. That said I'm happy to add them if you really feel those tests _aren't_ redundant.  Can you give me a sense of how the depth of testing you want for each public method?

**Comment by mreinstein** [comment]
I have significantly underestimated the complexity of making this change. :(

**Pull request closed without merging**